src/towncrier/_writer.py

```diff
@@ -8,7 +8,22 @@
 
 from __future__ import annotations
 
+import sys
+
 from pathlib import Path
+from typing import Any
+
+
+if sys.version_info < (3, 10):
+
+    def _newline_write_text(path: Path, content: str, **kwargs: Any) -> None:
+        with path.open("w", **kwargs) as strm:  # pragma: no cover
```

Comment:
Is this an issue of line coverage or branch coverage ?

```suggestion
        with path.open("w", **kwargs) as strm:  # pragma: no branch
```